src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -191,24 +203,38 @@ module.exports = function<T, P, I, TI, C, CX>(
     return workInProgress.child;
   }
 
-  function updateClassComponent(current : ?Fiber, workInProgress : Fiber) {
+  function updateClassComponent(current : ?Fiber, workInProgress : Fiber, priorityLevel : PriorityLevel) {
     let shouldUpdate;
     if (!current) {
       if (!workInProgress.stateNode) {
         // In the initial pass we might need to construct the instance.
         constructClassInstance(workInProgress);
-        mountClassInstance(workInProgress);
+        mountClassInstance(workInProgress, priorityLevel);
         shouldUpdate = true;
       } else {
         // In a resume, we'll already have an instance we can reuse.
-        shouldUpdate = resumeMountClassInstance(workInProgress);
+        shouldUpdate = resumeMountClassInstance(workInProgress, priorityLevel);
       }
     } else {
-      shouldUpdate = updateClassInstance(current, workInProgress);
+      shouldUpdate = updateClassInstance(current, workInProgress, priorityLevel);
     }
-    if (!shouldUpdate) {
+
+    // Schedule side-effects
+    if (shouldUpdate) {
+      workInProgress.effectTag |= Update;
+    } else {
+      // If an update was already in progress, we should schedule an Update
+      // effect even though we're bailing out, so that cWU/cDU are called.
```

Comment:
This doesn't affect CWU. Will need to come back to this to ensure I understand it.